MapboxCoreNavigation/RouteController.swift

```diff
@@ -586,7 +581,9 @@ extension RouteController: CLLocationManagerDelegate {
                 self.rawLocation = lastLocation
                 
                 // Check for a tunnel intersection at the current step we found the bad location update.
-                checkForTunnelIntersection(at: lastLocation, for: manager)
+                if tunnelSimulationEnabled {
```

Comment:
@bsudekum we should probably let the `TunnelIntersectionManager` maintain the status of the `tunnelSimulationEnabled`. This will definitely come in handy for future maintenance efforts.